lib/index.js

```diff
@@ -265,10 +265,12 @@ JobQueue.prototype.shutdown = function(callback) {
 
   var pend = new Pend();
   pend.go(function(cb) {
-    self.redisClient.quit(function(err, result) {
-      if (err) {
-        self.emit('error', err);
-      }
+    self.redisClient.quit(function(err) {
+        /* We want to kill this instance of the redis client, so at at this point we
+         * are not interested in what connection errors Redis comes up with.
+         */
+
+        self.redisClient.end();
```

Comment:
If you by 'original code' mean the calling client code, then it is simply by your API design. The redis client is an internal property of your API. As a client I am simply supplying you with `{redisConfig : { port : ..., host : ..., pass }}`, not an actual client, so the control of the redis client is an internal detail of your implementation on which I cannot rely. You might rename it in later versions, or do something entirely different.

If you on the other hand asked for client to supply you with redis clients in the constructor, then we would have a direct handle to the redis client and we could configure that as we pleased. 

So in conclusion: as the redis client is your responsibility, then you also need to make that it does not cause problems for the clients when the client is using your API to shut down the job queue.